plugins/USBPrinting/USBPrinterOutputDeviceManager.py

```diff
@@ -128,8 +128,12 @@ def getInstance(cls, engine = None, script_engine = None):
         return USBPrinterOutputDeviceManager._instance
 
     def _getDefaultFirmwareName(self):
-        machine_instance = Application.getInstance().getMachineManager().getActiveMachineInstance()
-        machine_type = machine_instance.getMachineDefinition().getId()
+        # Detecting id of the current machine
+        machine_id = Application.getInstance().getGlobalContainerStack().getBottom().id
```

Comment:
Probably the easiest yes. We usually simply return a default value in these situations and let the caller code deal with it.